### PR TITLE
64 bit compile patch

### PR DESCRIPTION
**Pull request by locusf**

### CODE REVIEW

**Comment by locusf** [comment]
Please wait for this to be complete, don't have a x86_64 compiler setup yet, will use OBS.

**Comment by matthewvogt** [comment]
LGTM

File: src/engine/memorytable.cpp [hunk 1]
```diff
@@ -338,7 +338,7 @@ quint32 MemoryTablePrivate::allocate(quint32 size, TableMetadata *table, bool in
 
     // Align the allocation so that the header is directly accessible
     quint32 allocationSize = static_cast<quint32>(requiredSpace(size));
```
Comment (SfietKonstantin) [inline src/engine/memorytable.cpp]: Maybe this should return a size_t, to avoid useless static_cast, and enforce 64 bits compatibility

**Comment by thp** [comment]
Can we get this merged?

**Comment by matthewvogt** [comment]
Actually, I'm not sure about this change.  What is the compile error it fixes?

**Comment by locusf** [comment]
This can be closed, it requires many more changes than this. Apparently the GCC used in obs creates type conflicts for unsigned long and since this software minds the types in the tests as well, it fails.

**Pull request closed without merging**